src/linear_algebra/basic.lean

```diff
@@ -358,6 +358,17 @@ instance linear_map_apply_is_add_group_hom (a : M) :
 
 end add_comm_group
 
+section nontrivial
+variables [semiring R] [add_comm_monoid M] [semimodule R M]
+
+instance [nontrivial M] : nontrivial (module.End R M) :=
+begin
+  obtain ⟨m, ne⟩ := (nontrivial_iff_exists_ne (0 : M)).mp infer_instance,
+  exact nontrivial_of_ne 1 0 (λ p, ne (linear_map.congr_fun p m)),
+end
+
+end nontrivial
```

Comment:
I think this would make more sense in `linear_map.lean`, since it doesn't need any of the submodule stuff developed in this file, and that's where `End` is defined.

Reply:
It needs the `1` and `0` on `module.End R M`, which are only developed in this file. I'll move it a little higher up, immediately after the definition of `1`.